### DISCUSSION TITLE
Is it possible to select the quality of a hls stream from code?

### DISCUSSION THREAD

**Discussion by thomaswagnerapa** [videojs/video.js#7928] [General]
Hi,

One of our customers uses the VideoJS, and wants to determine on the website itself, with which quality of his livestream is used. So, for example, with a smaller video window the lowest quality, with fullscreen the highest. I have now searched for some time, but unfortunately found nothing suitable. Can maybe someone here help me please?

Thank you very much and best regards
Thomas

**Comment by gjanblaszczyk** [comment]
Hi Thomas.
Yes, you can do that via video.js plugin called `videojs-contrib-quality-levels`. Please check the details here: https://github.com/videojs/videojs-contrib-quality-levels

**Reply by gkatsev** [reply]
There's also https://github.com/FoxCouncil/videojs-max-quality-selector for a UI.

Also, worth noting, this won't work on Safari by default and won't ever work on iPhones.
